roles/openshift_storage_glusterfs/tasks/glusterfs_deploy.yml

```diff
@@ -0,0 +1,106 @@
+---
+- assert:
+    that: "openshift_storage_glusterfs_nodeselector.keys() | count == 1"
+    msg: Only one GlusterFS nodeselector key pair should be provided
+
+- assert:
+    that: "groups.oo_glusterfs_to_config | count >= 3"
+    msg: There must be at least three GlusterFS nodes specified
+
+- name: Delete pre-existing GlusterFS resources
+  oc_obj:
+    namespace: "{{ openshift_storage_glusterfs_namespace }}"
+    kind: "template,daemonset"
+    name: glusterfs
+    state: absent
+  when: openshift_storage_glusterfs_wipe
+
+- name: Unlabel any existing GlusterFS nodes
+  oc_label:
+    name: "{{ item }}"
+    kind: node
+    state: absent
+    labels: "{{ openshift_storage_glusterfs_nodeselector | oo_dict_to_list_of_dict }}"
+  with_items: "{{ groups.all }}"
+  when: openshift_storage_glusterfs_wipe
+
+- name: Delete pre-existing GlusterFS config
+  file:
+    path: /var/lib/glusterd
+    state: absent
+  delegate_to: "{{ item }}"
+  with_items: "{{ groups.oo_glusterfs_to_config }}"
+  when: openshift_storage_glusterfs_wipe
+
+- name: Get GlusterFS storage devices state
+  command: "pvdisplay -C --noheadings -o pv_name,vg_name {% for device in hostvars[item].glusterfs_devices %}{{ device }} {% endfor %}"
+  register: devices_info
+  delegate_to: "{{ item }}"
+  with_items: "{{ groups.oo_glusterfs_to_config }}"
+  failed_when: False
+  when: openshift_storage_glusterfs_wipe
+
+- name: Clear GlusterFS storage device contents
+  shell: "{% for line in item.stdout_lines %}{% set fields = line.split() %}{% if fields | count > 1 %}vgremove -fy {{ fields[1] }}; {% endif %}pvremove -fy {{ fields[0] }}; {% endfor %}"
```

Comment:
I'm not sure it would make things any easier, unless using the `command` module provides any tangible benefit here. Also we'd be passing multi-line input into the script which could be hard to properly test.